packages/forma-36-react-components/src/components/Dropdown/DropdownContainer/DropdownContainer.tsx

```diff
@@ -1,252 +1,114 @@
+    const setReference = refCallback as React.Dispatch<
+      React.SetStateAction<HTMLElement | null>
+    >;
+    const dropdown = useRef<HTMLDivElement | null>(null);
+    const portalTarget = useRef<HTMLDivElement>(document.createElement('div'));
+    const classNames = cn(className, styles['DropdownContainer']);
+    useOnClickOutside(dropdown, () => {
+      if (isOpen && onClose) {
+        onClose();
       }
-    }
-  };
-
-  handleOverflow = (overflowAt: string) => {
-    if (!this.props.isAutoalignmentEnabled) {
-      return;
-    }
-    if (overflowAt === this.lastOverflowAt) {
-      return;
-    }
-
-    const resolutions = {
-      right: {
-        'bottom-left': 'bottom-right',
-        'top-left': 'top-right',
-        right: 'left',
-      },
-      left: {
-        'bottom-right': 'bottom-left',
-        'top-right': 'top-left',
-        left: 'right',
-      },
-      top: {
-        'top-left': 'bottom-left',
-        'top-right': 'bottom-right',
-      },
-      bottom: {
-        'bottom-left': 'top-left',
-        'bottom-right': 'top-right',
-      },
-    };
-    const currentPosition = this.state.position;
-    const resolution = resolutions[overflowAt][currentPosition];
-    if (resolution) {
-      this.setState(
-        {
-          position: resolution,
-        },
-        () => {
-          this.lastOverflowAt = overflowAt;
-        },
-      );
-    }
-  };
+    });
 
-  calculatePosition = () => {
-    const { anchorDimensionsAndPositon } = this.props;
-    const { dropdownDimensions, position } = this.state;
+    useEffect(() => {
+      document.body.appendChild(portalTarget.current);
 
-    if (!anchorDimensionsAndPositon || !dropdownDimensions) {
-      return false;
-    }
-
-    switch (position) {
-      case 'bottom-left':
-        return {
-          top:
-            anchorDimensionsAndPositon.top + anchorDimensionsAndPositon.height,
-          left: anchorDimensionsAndPositon.left,
-        };
-      case 'top-left':
-        return {
-          bottom: window.innerHeight - anchorDimensionsAndPositon.top,
-          left: anchorDimensionsAndPositon.left,
-        };
-      case 'bottom-right':
-        return {
-          top:
-            anchorDimensionsAndPositon.top + anchorDimensionsAndPositon.height,
-          left:
-            anchorDimensionsAndPositon.left -
-            (dropdownDimensions.width - anchorDimensionsAndPositon.width),
-        };
-      case 'top-right':
-        return {
-          bottom: window.innerHeight - anchorDimensionsAndPositon.top,
-          left:
-            anchorDimensionsAndPositon.left -
-            (dropdownDimensions.width - anchorDimensionsAndPositon.width),
-        };
-    }
-  };
-
-  getSubmenuClassNames = () =>
-    cn(
-      styles['DropdownContainer__submenu'],
-      styles[`DropdownContainer__container-position--${this.state.position}`],
-    );
+      return () => {
+        document.body.removeChild(portalTarget.current);
```

Comment:
Maybe we should refactor the Tooltip to use Portals as well. But In another PR of course 😄